marquez_client/client.py

```diff
@@ -80,16 +97,204 @@ def list_namespaces(self, limit=None, offset=None):
             }
         )
 
-    def create_job(self, job_name, job_type, location=None, input_dataset=None,
-                   output_dataset=None, description=None, context=None,
-                   namespace_name=None):
+    # Sources API
+    def create_source(self, source_name, source_type, connection_url,
+                      description=None):
+        if not source_name:
+            raise ValueError('source_name must not be None')
+        if len(source_name) > _SOURCE_LENGTH_LIMIT:
+            raise ValueError('source_name length is {0}, must be <= {1}',
+                             len(source_name), _SOURCE_LENGTH_LIMIT)
+
+        if not connection_url:
+            raise ValueError('connection_url must not be None')
+
+        if not isinstance(source_type, SourceType):
+            raise ValueError(f'source_type must be instance of SourceType')
+
+        payload = {
+            'type': source_type.value,
+            'connectionUrl': connection_url
+        }
+
+        if description:
+            payload['description'] = description
+
+        return self._put(self._url('/sources/{0}', source_name),
+                         payload=payload)
+
+    def get_source(self, source_name):
+        if not source_name:
+            raise ValueError('source_name must not be None')
+        if len(source_name) > _SOURCE_LENGTH_LIMIT:
+            raise ValueError('source_name length is {0}, must be <= {1}',
+                             len(source_name), _SOURCE_LENGTH_LIMIT)
```

Comment:
Great error messages. That improves user experience.
Possibly factor out the length check logic into a reusable function.

Reply:
done.